src/AzSK.ADO/Framework/Abstracts/ADOSVTCommandBase.ps1

```diff
@@ -45,6 +45,14 @@ class ADOSVTCommandBase: SVTCommandBase {
         if ([ContextHelper]::IsOAuthScan) {
             $this.PublishCustomMessage("The OAuth-based scan feature is in preview. Results for some controls may not match the regular scan results.", [MessageType]::Warning);
         }
+        try 
+        {
+            # Checking if user has graph access or not before starting control evaluation.
+            [IdentityHelpers]::CheckGraphAccess();
```

Comment:
cant we add if(![IdentityHelpers]::hasGraphAccess) before calling this method, if alaredy evaluated and got true then no need to evaluate in next scan

Reply:
This method will be evaluated only once at the staring of the scan.